### PR TITLE
assignment 4

### PR DESCRIPTION
**Pull request by ander-ros**

### CODE REVIEW

File: Assignment4/ander-ros-000731587/task06.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+# TO DO
+g.add((ns.University, RDF.type, RDFS.Class))
+# Visualize the results
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.2: Add "Researcher" as a subclass of "Person"**"""
+
+# TO DO
+g.add((ns.Researcher, RDF.type, RDFS.Class))
```
Comment (SerPablo) [inline Assignment4/ander-ros-000731587/task06.py:44]: Researcher and person are already defined

File: Assignment4/ander-ros-000731587/task07.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+"""First let's read the RDF file"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
```
Comment (SerPablo) [inline Assignment4/ander-ros-000731587/task07.py:23]: All the exercises have to be done in SPARQL and RDFLIb

File: Assignment4/ander-ros-000731587/task07.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+"""**TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**"""
+
+# TO DO
+from rdflib.plugins.sparql import prepareQuery
+q1 = prepareQuery("""
+    SELECT ?subclass
+    WHERE {
+        ?subclass rdfs:subClassOf* ns:LivingThing .
```
Comment (SerPablo) [inline Assignment4/ander-ros-000731587/task07.py:31]: subClassOf+

File: Assignment4/ander-ros-000731587/task07.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+"""**TASK 7.3: List all individuals of "Person" or "Animal" and all their properties including their class with RDFLib and SPARQL. You do not need to list the individuals of the subclasses of person**
+
+"""
+
+# TO DO
+query = prepareQuery("""
+    SELECT ?individual ?property ?class
+    WHERE {
+        {?individual a/rdfs:subClassOf* ns:Person}  UNION {?individual a/rdfs:subClassOf* ns:Animal.}
```
Comment (SerPablo) [inline Assignment4/ander-ros-000731587/task07.py:61]: It's correct but maybe is better to not show always the Subject repeated

File: Assignment4/ander-ros-000731587/task07.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        ?subClass rdfs:subClassOf ns:Person.
+      }
+      UNION
+      {
+        ?peopleName rdf:type ns:Person.
+      }
+    }
+  }
+  """,
+  initNs = { "vcard": Namespace("http://www.w3.org/2001/vcard-rdf/3.0#").replace("#","/"), "rdfs":RDFS, "ns":NS, "foaf":FOAF}
```
Comment (SerPablo) [inline Assignment4/ander-ros-000731587/task07.py:91]: Only Persons, check UPPERCASE of NS

**Review by SerPablo — COMMENTED** [review]
Review comments in task 6,7. Remove file of task 9

**Comment by anaigmo** [comment]
Submitted and mostly fixed, no more PRs allowed

**Pull request merged**